docs/fsspec.rst

```diff
@@ -0,0 +1,93 @@
+Fsspec Filesystem
```

Comment:
For now went with a recipe-based approach instead of documenting every option one-by-one